src/Components/Components/src/BindConverter.cs

```diff
@@ -0,0 +1,1047 @@
+        internal static BindParser<DateTimeOffset?> ConvertToNullableDateTimeOffset = ConvertToNullableDateTimeOffsetCore;
+        internal static BindParserWithFormat<DateTimeOffset?> ConvertToNullableDateTimeOffsetWithFormat = ConvertToNullableDateTimeOffsetCore;
+
+        private static bool ConvertToDateTimeOffsetCore(object obj, CultureInfo culture, out DateTimeOffset value)
+        {
+            return ConvertToDateTimeOffsetCore(obj, culture, format: null, out value);
+        }
+
+        private static bool ConvertToDateTimeOffsetCore(object obj, CultureInfo culture, string format, out DateTimeOffset value)
+        {
+            var text = (string)obj;
+            if (string.IsNullOrEmpty(text))
+            {
+                value = default;
+                return false;
+            }
+
+            if (format != null && DateTimeOffset.TryParseExact(text, format, culture ?? CultureInfo.CurrentCulture, DateTimeStyles.None, out var converted))
+            {
+                value = converted;
+                return true;
+            }
+            else if (format == null && DateTimeOffset.TryParse(text, culture ?? CultureInfo.CurrentCulture, DateTimeStyles.None, out converted))
+            {
+                value = converted;
+                return true;
+            }
+
+            value = default;
+            return false;
+        }
+
+        private static bool ConvertToNullableDateTimeOffsetCore(object obj, CultureInfo culture, out DateTimeOffset? value)
+        {
+            return ConvertToNullableDateTimeOffsetCore(obj, culture, format: null, out value);
+        }
+
+        private static bool ConvertToNullableDateTimeOffsetCore(object obj, CultureInfo culture, string format, out DateTimeOffset? value)
+        {
+            var text = (string)obj;
+            if (string.IsNullOrEmpty(text))
+            {
+                value = default;
+                return true;
+            }
+
+            if (format != null && DateTimeOffset.TryParseExact(text, format, culture ?? CultureInfo.CurrentCulture, DateTimeStyles.None, out var converted))
+            {
+                value = converted;
+                return true;
+            }
+            else if (format == null && DateTimeOffset.TryParse(text, culture ?? CultureInfo.CurrentCulture, DateTimeStyles.None, out converted))
+            {
+                value = converted;
+                return true;
+            }
+
+            value = default;
+            return false;
+        }
+
+        private static bool ConvertToEnum<T>(object obj, CultureInfo culture, out T value) where T : struct, Enum
+        {
+            var text = (string)obj;
+            if (string.IsNullOrEmpty(text))
+            {
+                value = default;
+                return true;
+            }
+
+            if (!Enum.TryParse<T>(text, out var converted))
```

Comment:
At some point we want to impose the rule that we only consider it parseable if it maps to a named value, right? Might this PR be a good opportunity to introduce that? I *hope* it's as simple as adding `&& Enum.IsDefined(typeof(T), converted)`.

I think we'd want that rule in before 3.0 ships otherwise it's (at least arguably) a breaking change later.

Reply:
Yeah, I agree that we should do this.